### ISSUE TITLE
Word min argument regression in Pyparsing 3.1.0

### ISSUE THREAD

**Issue by volans-** [pyparsing/pyparsing#502]
### Summary

It seems to me that there is a regression in Pyparsing 3.1.0 at least for the `min` argument of the `Word` class. I quickly tested the `max` one that seems to work but I didn't test all the combinations between `min`, `max` and `exact`.

### Versions

* Python 3.11.3
* Pyaprsing 3.1.0

### Repro code

```python
>>> import pyparsing
>>> grammar = pyparsing.Word('abc', min=2)('key')
>>> grammar.parseString('a'.strip(), parseAll=True)
ParseResults(['a'], {'key': 'a'})
```

With version `3.0.9` that raises as expected:
```python
pyparsing.exceptions.ParseException: Expected W:(abc){2,...}, found end of text  (at char 1), (line:1, col:2)
```

**Comment by ptmcg** [comment]
`git bisect` points to this change: https://github.com/pyparsing/pyparsing/commit/7ec34f497ebc9cbfd51ffeb5cce569133eb7c3c1 , thanks for submitting an easy-to-reproduce case.

**Comment by ptmcg** [comment]
I was so focused on fixing max, I forgot to check min. Checking in the fix to this shortly.

**Comment by ptmcg** [comment]
Fixed in 3.1.1, released today
